### PR TITLE
upcoming: [M3-7777, M3-7778] - Improve Billing & Account Restricted User Experience

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
For the three account options: "Make a Payment," "Update Billing Contact Information," and "Add Payment Method," access will be disabled for restricted users or child accounts. Furthermore, parent account users will not have the ability to update the company name.

>[!note]
We have separate tickets for additional E2E test coverage [M3-7797] 👍. 

## Changes  🔄
List any change relevant to the reviewer.
- Removed `getDisabledTooltipText` util and replaced it with our existing `useRestrictedGlobalGrantCheck` and `getRestrictedResourceText` util.

## Target release date 🗓️
3/4/2024

## Preview 📷
| Before  | After: User Type = Default   | After: User Type = Child | After: User Type = Parent |
| ------- | ------- |------- |------- |
| Screenshot 2024-02-16 at 12 00 46 PM | Screenshot 2024-02-16 at 12 00 24 PM |Screenshot 2024-02-16 at 2 37 22 PM | Screenshot 2024-02-20 at 5 30 55 PM |
| Screenshot 2024-02-16 at 12 26 14 PM | Screenshot 2024-02-16 at 12 48 53 PM |Screenshot 2024-02-16 at 2 37 40 PM |
| Screenshot 2024-02-16 at 12 24 05 PM | Screenshot 2024-02-16 at 12 23 33 PM |Screenshot 2024-02-16 at 2 38 13 PM |

### Prerequisites
(How to setup test environment)
- Log into two accounts side by side:
   - An unrestricted admin user account: full access
   - A restricted user account (use Incognito for this)
       - Start with `Read Only` for everything

### Reproduction steps
(How to reproduce the issue, if applicable)
- Observe that before changes, a user can access and even begin filling out form information before eventually getting denied by the API. We want to avoid this.

### Verification steps
(How to verify changes)
- Test these flows as a restricted user with 'read only' access
- Then turn on MSW and Parent/Child feature flag
    - As a parent, observe that you cannot edit company name
- Set `user_type` to `child` in [serverHandlers](https://github.com/linode/manager/blob/develop/packages/manager/src/mocks/serverHandlers.ts#L554)  
    - Observe that you cannot access any of these drawers, and if you navigate via the URL's you get appropriate error notices with disabled fields

1. http://localhost:3000/account/billing/edit
2. http://localhost:3000/account/billing/add-payment-method 
3. http://localhost:3000/account/billing/make-payment

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Billing/BillingPanels/PaymentInfoPanel/AddPaymentMethodDrawer/AddPaymentMethodDrawer.tsx [hunk 1]
```diff
@@ -110,21 +131,23 @@ export const AddPaymentMethodDrawer = (props: Props) => {
...
-            >
-              <GooglePayChip
-                disabled={disabled}
-                onClose={onClose}
-                renderError={renderError}
-                setMessage={setMessage}
-                setProcessing={setIsProcessing}
-              />
-            </Grid>
+            {!isReadOnly && (
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Billing/BillingPanels/PaymentInfoPanel/AddPaymentMethodDrawer/AddPaymentMethodDrawer.tsx:134]: Same here. No need to load/render this button

**Comment by jdamore-linode** [comment]
@jaalah-akamai Pushed a couple commits that'll fix 3 of the test failures (basically just amounted to wording updates in the tests). However, 2 of the failures are caused by app crashes -- I think it has to do with `useRestrictedGlobalGrantCheck` tripping up when the API grants endpoint responds with a 204 and doesn't contain any response data.

https://github.com/linode/manager/blob/2b8d1c803266bf6d3efa680f8af569cc0f1eea8d/packages/manager/src/hooks/useRestrictedGlobalGrantCheck.ts#L16-L31

I quickly tried replacing `grants?.global[globalGrantType]` with `grants?.global?.[globalGrantType]` and that seemed to allow the tests to pass, but I didn't look real closely in case better/more thorough logic is necessary. We might want a test for this case in `useRestrictedGlobalGrantCheck.test.ts` at some point, too

**Review by mjac0bs — COMMENTED** [review]
The UI changes look good! Confirmed:
- parent users cannot edit the company name
- child users and regular restricted users see a disabled `Make a Payment` button with a tooltip message
- child users and regular restricted users see an error notice and disabled fields in Billing Contact, Make a Payment, and Add Payment Method drawers, if accessed via URL

Still seeing the following e2e failures locally, though Joe may already be on it as I review this:

`account/account-cancellation.spec.ts`
- Renders an "Oh Snap!" error for test `restricted users cannot cancel an account`

`account/account-login-history.spec.ts`
- Renders an "Oh Snap!" error for test `restricted users cannot view login history`

**Comment by jaalah-akamai** [comment]
@jdamore-linode Yep 100% good catch 👍

File: packages/manager/src/components/AccountActivation/AccountActivationError.tsx [hunk 2]
```diff
@@ -23,10 +23,12 @@ const AccountActivationError = (props: CombinedProps) => {
...
   return (
     <React.Fragment>
-      {getAPIErrorOrDefault(
-        props.errors,
-        'Your account is not yet activated. Please reach out to support@linode.com for more information'
-      )[0].reason}
+      {
+        getAPIErrorOrDefault(
+          props.errors,
+          'Your account is not yet activated. Please reach out to support@linode.com for more information'
```
Comment (carrillo-erik) [inline packages/manager/src/components/AccountActivation/AccountActivationError.tsx:29]: Might not be a big deal in this case but semantically speaking, should the email address take an anchor tag?
`<a href="mailto:support@linode.com">support@linode.com</a>`

**Review by carrillo-erik — APPROVED** [review]

**Comment by carrillo-erik** [comment]
Nice work cleaning up some files and making the changes to the UI. Looks good to me and did not see any UI regressions. I was able to verify the changes locally.

**Pull request merged**